packages/metro-tree-shaking/src/index.js

```diff
@@ -86,7 +88,7 @@ async function removeUnUsedExports(
   const globalDependencies = graph.dependencies;
   // analyze AST
   for (const [absolutePath, module] of globalDependencies.entries()) {
-    if (treeShakingIgnore(absolutePath)) {
+    if (treeShakingIgnore(absolutePath) || module.importee.sideEffect) {
```

Comment:
`sideEffects` (it should be plural, see https://webpack.js.org/guides/tree-shaking/)

Reply:
`sideEffects` is for npm package(package usually has many files)，but in this line, it used to single file, so it is singular. I will merge `package.json's sideEffects` https://webpack.js.org/guides/tree-shaking/  to single file's sideEffect when i'm free

Reply:
OK no worries! My bad.